### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by joefitter**
* added deps - codecept, express-session, lodash & WebDriverIO
* added base config
* added helpers - session and navigation from hmpo-enquiries
* added step extensions
* added setup/teardown
* wrote a bunch of tests

### CODE REVIEW

**Comment by joefitter** [comment]
@JoeChapman @easternbloc @daniel-ac-martin @danielepolencic @bendiggle mind giving this a once over? We are using CodeceptJS for acceptance testing in hmpo-enquiries and ukvi complaints

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,674 @@
+GNU GENERAL PUBLIC LICENSE
+   Version 3, 29 June 2007
```
Comment (easternbloc) [inline LICENSE]: we do GPLv2 not v3 afaik

Comment (joefitter) [inline LICENSE]: Just copied from bootstrap so that must be WRONG 😛

Comment (joefitter) [inline LICENSE]: @JoeChapman ^

**Comment by JoeChapman** [comment]
Is this hof-acceptance or sof-acceptance?

**Comment by joefitter** [comment]
Depends what people think? @daniel-ac-martin @danielepolencic @gavboulton @easternbloc? Personally I like the idea of standardising tests across hof applications, not just within SO. Happy to rename to SOF if we just wanna keep it to ourselves

**Comment by easternbloc** [comment]
can we make it so-acceptance till we discuss it across other teams

**Comment by danielepolencic** [comment]
image

**Comment by joefitter** [comment]
Haha. Renamed to SO Acceptance for now

**Comment by easternbloc** [comment]
best repo name EVER

**Comment by joefitter** [comment]
disappointed the meme generator didn't use purple comic sans by default

**Comment by daniel-ac-martin** [comment]
Has this stuff been compared to chimp? What did people think?

**Comment by joefitter** [comment]
@gxxm did a spike comparing different node acceptance testing frameworks

**Comment by joefitter** [comment]
he is on hols though

**Comment by joefitter** [comment]
@JoeChapman @bendiggle @gxxm @easternbloc as this is currently "SO Acceptance", mind merging it so we can use it for UKVI and HMPO Enquiries?

**Comment by bendiggle** [comment]
👍

**Comment by gxxm** [comment]
:thumbsup: :thumbsup:

**Pull request merged**